### PR TITLE
Adgroup Apis

### PR DESCRIPTION
**Pull request by minsikzzang**
Adgroup Apis

Related Documents
- http://naver.github.io/searchad-apidoc/#/tags/Adgroup

### CODE REVIEW

File: lib/naver/searchad/api/errors.rb [hunk 1]
```diff
@@ -49,7 +49,9 @@ class MissingRequiredAttributeError < RequestError; end
 
       class InvalidRequestError < RequestError; end
 
-      class CampaignAlreadyExistError < RequestError; end
+      class DuplicatedCampaignNameError < RequestError; end
+
+      class DuplicatedAdgroupNameError < RequestError; end
```
Comment (chiting) [inline lib/naver/searchad/api/errors.rb]: What about the idiom `DuplicatedAdgroupNameError = Class.new(RequestError)`?
I like both but for an empty class that looks to suit better, what do you think?

File: lib/naver/searchad/api/adgroup/service.rb [hunk 2]
```diff
@@ -5,35 +5,55 @@ module Adgroup
         class Service < Naver::Searchad::Api::Core::BaseService
 
           def initialize
-            super('https://api.naver.com/', 'ncc/adgroups/')
+            super('https://api.naver.com/', 'ncc/')
```
Comment (chiting) [inline lib/naver/searchad/api/adgroup/service.rb]: Can we put this hard-coded string in a constant?

File: lib/naver/searchad/api/core/base_service.rb [hunk 3]
```diff
@@ -55,6 +55,13 @@ def apply_command_defaults(command)
             # To be implemented by subclasses
           end
 
+          def validates_presence_of(fields, object)
+            fields.each do |key|
+              raise MissingRequiredAttributeError.new(
```
Comment (chiting) [inline lib/naver/searchad/api/core/base_service.rb]: As it's already too long for one line, this one liner doesn't make much sense :D
```
unless object.key?(key)
  raise MissingRequiredAttributeError
    .new("Require #{key} attribute in object")
end
```

File: lib/naver/searchad/api/adgroup/service.rb (hunk 2 above)
Comment (minsikzzang) [inline lib/naver/searchad/api/adgroup/service.rb]: even I put them in constants, they will be used only once. What about adding `freeze` to them? at least, they will be immutable :)

File: lib/naver/searchad/api/errors.rb (hunk 1 above)
Comment (minsikzzang) [inline lib/naver/searchad/api/errors.rb]: prefer `Class.new(RequestError)`, too :)

File: lib/naver/searchad/api/adgroup/service.rb (hunk 2 above)
Comment (chiting) [inline lib/naver/searchad/api/adgroup/service.rb]: Oh yeah I mean, it's because it's hardcoded that we should at least abstract it (like for your specs :)

Comment (minsikzzang) [inline lib/naver/searchad/api/adgroup/service.rb]: reason why I put them to constants in spec, I have to reuse them across many test cases. I don't think it is necessary here 👍 thanks for your comment anyway

**Pull request merged**